nixos/modules/services/misc/matrix-appservice-irc.nix

```diff
@@ -0,0 +1,121 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.matrix-appservice-irc;
+
+  defaultConfig = {
+    database = {
+      engine = "nedb";
+      connectionString = "nedb://var/lib/matrix-appservice-irc/data";
+    };
+    ircService.passwordEncryptionKeyPath = "/var/lib/matrix-appservice-irc/passkey.pem";
+  };
+  configFile = pkgs.writeText "matrix-appservice-irc.yml" (builtins.toJSON (recursiveUpdate cfg.config defaultConfig));
+  bin = "${pkgs.matrix-appservice-irc}/bin/matrix-appservice-irc";
+  registrationFile = "/var/lib/matrix-appservice-irc/registration.yml";
+in {
+  options.services.matrix-appservice-irc = with types; {
+    enable = mkEnableOption "the Matrix/IRC bridge";
+
+    port = mkOption {
+      type = port;
+      description = "The port to listen on";
+      default = 8009;
+    };
+
+    needBindingCap = mkOption {
+      type = bool;
+      description = "Whether the daemon needs to bind to ports less than 1024 (e.g. for the ident service)";
+      default = false;
```

Comment:
Shouldn't the default depend on the selected port? `default = cfg.port < 1024`

Reply:
No, because the option is unrelated to `cfg.port`. It's required when matrix-appservice-irc is supposed to run an identd server on 113/tcp.

https://github.com/matrix-org/matrix-appservice-irc/blob/master/config.sample.yaml#L367-L384